cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -93,23 +94,28 @@ void copy_buffer_data(string_policy const& buff, string_policy& new_buff)
 template <class string_policy>
 void column_buffer_base<string_policy>::create_with_mask(size_type _size,
                                                          cudf::mask_state null_mask_state,
+                                                         bool memset_data,
                                                          rmm::cuda_stream_view stream,
                                                          rmm::device_async_resource_ref mr)
 {
   size = _size;
   _mr  = mr;
 
   switch (type.id()) {
-    case type_id::STRING: static_cast<string_policy*>(this)->allocate_strings_data(stream); break;
+    case type_id::STRING:
+      static_cast<string_policy*>(this)->allocate_strings_data(memset_data, stream);
+      break;
 
     // list columns store a buffer of int32's as offsets to represent
     // their individual rows
-    case type_id::LIST: _data = create_data(data_type{type_id::INT32}, size, stream, _mr); break;
+    case type_id::LIST:
+      _data = create_data(data_type{type_id::INT32}, size, memset_data, stream, _mr);
```

Comment:
This may not be correct with large strings, which use 64 bit offsets. @davidwendt Can you weigh in here?

Reply:
Lists columns offsets are only 32-bit (INT32). Strings columns offsets can be either INT32 or INT64 depending on the size of the data.

Reply:
Argh. I must have misread "list" as "string." Apologies. Either way, can we use `data_type{type_to_id<size_type>()}` instead of `data_type{type_id::INT32}`?

Reply:
I think INT32 would be more correct since changing `size_type` would no longer make this correct.

Reply:
We use size_type and not int32 for lists offsets across libcudf, in my understanding. https://github.com/rapidsai/cudf/blob/8068a2d616b6647bcd80720a2c24af858cbffd2d/cpp/include/cudf/lists/lists_column_view.hpp#L76

Reply:
Yes, those will have to change if we ever change `size_type` since offsets could still be `INT32` regardless of `size_type`. We get away with this since `size_type` always maps to `INT32` for now. I think we'll be safe for the foreseeable future using `size_type` for list offsets. But I posit the `INT32/int32_t` would be more clear/correct.